connectors/bitbucket-server/src/main/resources/static/discovery/metadata.hal

```diff
@@ -0,0 +1,9 @@
+{
+  "authorization_header": "x-bitbucket-server-authorization",
+  "backend_base_url_header": "x-bitbucket-server-base-url",
+  "fields": {
+    "pr_email_subject": {
+      "regex": "(([a-zA-Z0-9]+)\/([a-zA-Z0-9-]+) - Pull request #([0-9]+):[ ])"
```

Comment:
Was this not a problem for you when you manually tested?  When I manually tested in the ios simulator, the client was sending in the `([0-9]+)` as the tokens.

Reply:
I'll try again in with the updated code in a few minutes.

Reply:
```
Req Body
--------------------
{
  "card_requests" : [
    {
      "name" : "BitbucketServer",
      "tokens" : {
        "pr_email_subject" : [
          "252",
          "251"
        ]
      },
      "connector_id" : "7f66ed5c-e990-419c-b384-8d9084ba0dbc"
    }
  ]
}
```

While I think the client should return the first regex group (but it isn't specified in our docs what to do when multiple regex groups), it currently isn't (it returns the last regex group).  We can fix this by excluding the inner groups:

```
((?:[a-zA-Z0-9]+)\/(?:[a-zA-Z0-9-]+) - Pull request #(?:[0-9]+):[ ])
```

This will break your unit test as you currently have it written (that test is currently requiring the regex in the metadata.hal that we send to the client to be the same regex to support the same thing you are doing with our regex in the code that pulls out project/repo/id -- specified in a java constant).

Reply:
The input in the ios example app that produced that (not correct) request:

```
* UFO/app-platform-server - Pull request #251: APF-730: Make misconfigured connectors unhealthy
* UFO/app-platform-server - Pull request #252: APF-767 Loosen up handling of connector error responses
```

Reply:
When you ignore the inner groups, it will produce:

```
Req Body
--------------------
{
  "card_requests" : [
    {
      "name" : "BitbucketServer",
      "tokens" : {
        "pr_email_subject" : [
          "UFO/app-platform-server - Pull request #252: ",
          "UFO/app-platform-server - Pull request #251: "
        ]
      },
      "connector_id" : "7f66ed5c-e990-419c-b384-8d9084ba0dbc"
    }
  ]
}
```

Reply:
Yup!!

Reply:
Please change only the metadata.hal's regex (leave your java constant regex as-is) from:

```
"regex": "(([a-zA-Z0-9]+)\/([a-zA-Z0-9-]+) - Pull request #([0-9]+):[ ])"
```

to

```
"regex": "((?:[a-zA-Z0-9]+)\/(?:[a-zA-Z0-9-]+) - Pull request #(?:[0-9]+):[ ])"
```

so it will fix the ios simulator.

Reply:
The regex in the metadata.hal is run by the client to produce that list of "pr_email_subject".  When you use postman, it is skipping what the client is doing with the metadata.hal.  This is fine most of the time, but like I say in the other comments, what is currently in the metadata.hal will cause the client to send in the wrong input (it only sends in the numbers).

You have to leave your java constant regex as it is for your java code to work, but the metadata.hal has to be changed for the client to work.  See the other comment for details.